### ISSUE TITLE
Can anyone give a simple example to load and run inference on GPT-2 model in JAVA?

### ISSUE THREAD

**Issue by AayushSameerShah** [microsoft/onnxruntime-inference-examples#269]
I have **exported the GPT-2 model** in the ONNX format but I am not able to find any examples to perform the inference in the Java. Please help and share some code.

Thank you 🙏🏻

**Comment by AayushSameerShah** [comment]
Hello, @madroidmaq 
Here is a related answer to my question on another repo, please have a look, probably it will help 😃

https://github.com/microsoft/onnxruntime/issues/16515#issuecomment-1612493671

**Comment by Craigacp** [comment]
There are Java ports/interfaces for Wordpiece/BPE tokenizers, we wrote one for the [BERT tokenizer](https://github.com/oracle/tribuo/blob/main/Util/Tokenization/src/main/java/org/tribuo/util/tokens/impl/wordpiece/WordpieceTokenizer.java) and did GPT-2 as well though it looks like we didn't get around to releasing that, and DJL wrote a JNI binding for both [HuggingFace Tokenizers](https://github.com/deepjavalibrary/djl/tree/master/extensions/tokenizers) and [SentencePiece](https://github.com/deepjavalibrary/djl/tree/master/extensions/sentencepiece). As mentioned in the linked article there are ONNX extension operations that perform tokenization, those can be used via ONNX Runtime and you can even bind them into the same ONNX model as the GPT model.

Sampling can be written in Java, or done using the ONNX extension operations mentioned in the link. If you get the logits back out you can normalize them into a probability distribution via a softmax and then run whatever sampling procedure you want.
